pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -227,7 +227,11 @@ public void addSegment(@Nonnull String segmentName, @Nonnull TableConfig tableCo
         return;
       }
       Schema schema = ZKMetadataProvider.getTableSchema(_propertyStore, _tableNameWithType);
-      Preconditions.checkNotNull(schema);
+      if (schema == null) {
```

Comment:
We should be failing table add if schema is null. I think there was a PR that was merged which fixes the table add API. In that case, this null condition should arise only if someone deletes a schema, right? Precondition check seems fine for that? Maybe I am missing something?